.github/workflows/nightly-emscripten.yml

```diff
@@ -0,0 +1,124 @@
+name: Emscripten nightly build + push to solc-bin
+
+on:
+  schedule:
+    # Run once a day, at midnight
+    - cron: '0 0 * * *'
+
+env:
+  TARGET_BRANCH: gh-pages
+  COMMITTER_NAME: emscripten nightly action
+  COMMITTER_EMAIL: builds@ethereum.org
+  GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+jobs:
+  build-emscripten-nightly:
+    runs-on: ubuntu-latest
+    outputs:
+      solidity-version: ${{ env.SOLIDITY_VERSION }}
+      nightly-version: ${{ env.NIGHTLY_VERSION }}
+      matching-nightlies-in-the-repo: ${{ env.MATCHING_NIGHTLIES_IN_THE_REPO }}
+
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          repository: 'ethereum/solidity'
+          ref: 'develop'
+          path: 'solidity/'
+
+      - name: Clone solc-bin repository without checking out a working copy
+        run: |
+          git clone --no-checkout --branch "$TARGET_BRANCH" "https://github.com/${GITHUB_REPOSITORY}.git" solc-bin/
+
+          # For some reason git stages all files for deletion when you use --no-checkout
+          cd solc-bin/
+          git reset HEAD --quiet
+
+      - name: Check if there's already a nightly with the same date or commit ID
+        run: |
+          cd solidity/
+          solidity_version=$("scripts/get_version.sh")
+          last_commit_timestamp=$(git log -1 --date=iso --format=%ad HEAD)
+          last_commit_date=$(date --date="$last_commit_timestamp" --utc +%Y.%-m.%-d)
+          last_commit_hash=$(git rev-parse --short=8 HEAD)
+
+          cd ../solc-bin/
+          matching_nightlies_in_the_repo="$(
+            git ls-files "bin/soljson-v${solidity_version}-nightly.${last_commit_date}+commit.*.js";
+            git ls-files "bin/soljson-v${solidity_version}-nightly.*+commit.${last_commit_hash}.js"
+          )"
+          nightly_version="v${solidity_version}-nightly.${last_commit_date}+commit.${last_commit_hash}"
+
+          echo "::set-env name=SOLIDITY_VERSION::${solidity_version}"
+          echo "::set-env name=NIGHTLY_VERSION::${nightly_version}"
+          echo "::set-env name=MATCHING_NIGHTLIES_IN_THE_REPO::${matching_nightlies_in_the_repo}"
```

Comment:
Can't we just fail the build if there is a matching nightly? This would also stop all subsequent steps from being executed.

Reply:
We could but then we would all be getting a notification about the failed job. I think it's better not to desensitize people to real failures by having spurious one by design :)

I wish there was some way to just stop the job but as far as I know currently there isn't. Though I've seen people complain about that on github's community forum so maybe it will change in the future.

Reply:
Ah I see! Can you document the reason for not failing?

Reply:
Done.